flaskoidc/__init__.py

```diff
@@ -76,65 +38,97 @@ def _before_request(self):
+                except Exception as ex:
+                    LOGGER.debug("Token not found in the database, use the one in the request")
+                    # Since this is a request coming from other service,
+                    # we will need to assign the token.
+                    self.auth_client.token = token
+
+        try:
+            self.auth_client.token
+        except Exception as ex:
+            LOGGER.exception("User not logged in, redirecting to auth")
+            LOGGER.exception(ex)
+            return redirect(url_for('logout', _external=True))
 
     def __init__(self, *args, **kwargs):
         super(FlaskOIDC, self).__init__(*args, **kwargs)
 
-        # Setup Session Database
-        _sql_db = SQLAlchemy(self)
-        self.config["SESSION_SQLALCHEMY"] = _sql_db
+        self.db = SQLAlchemy(self)
+        _provider = self.config.get('OIDC_PROVIDER').lower()
 
-        # Setup Session Store, that will hold the session information
-        # in database. OIDC by default keep the sessions in memory
-        _session = Session(self)
-        _session.app.session_interface.db.create_all()
+        if _provider not in _CONFIGS.keys():
+            LOGGER.info(f"""
+            [flaskoidc Notice] I have not verified the OIDC Provider that you have 
+            selected i.e., "{_provider}" with this package yet. 
+            If you encounter any issue while using this library with "{_provider}",
+            please do not hesitate to create an issue on Github. (https://github.com/verdan/flaskoidc)
+            """)
 
-        # Initiate OpenIDConnect using the SQLAlchemy backed session store
-        _oidc = CustomOpenIDConnect(self, SessionCredentialStore())
-        self.oidc = _oidc
+        with self.app_context():
+            from flaskoidc.models import OAuth2Token, _fetch_token, _update_token
+            self.db.create_all()
+
+            oauth = OAuth(
+                self,
+                fetch_token=_fetch_token,
+                update_token=_update_token
+            )
+
+            self.auth_client = oauth.register(
+                name=_provider,
+                server_metadata_url=self.config.get('CONFIG_URL'),
+                client_kwargs={
+                    'scope': self.config.get('OIDC_SCOPES'),
+                },
+                **_CONFIGS.get(_provider) if _CONFIGS.get(_provider) else {}
+            )
 
         # Register the before request function that will make sure each
         # request is authenticated before processing
         self.before_request(self._before_request)
 
+        def unauthorized_redirect(err):
+            LOGGER.info("Calling the 401 Error Handler. 'unauthorized_redirect'")
+            return redirect(url_for('logout', _external=True))
+
+        self.register_error_handler(401, unauthorized_redirect)
+
         @self.route('/login')
         def login():
-            return redirect('/')
+            redirect_uri = url_for('auth', _external=True)
```

Comment:
should this be `REDIRECT_URI`?

Reply:
Nope, REDIRECT_URI is for the OIDC Provider to resirect to exchange token. 
We can have one LOGOUT_REDIRECT_URI though. Will add one in the next version.